### PR TITLE
ISPN-14638 RPOP and LPOP commands

### PR DESCRIPTION
**Pull request by karesti**
https://issues.redhat.com/browse/ISPN-14638

### CODE REVIEW

**Comment by karesti** [comment]
@wburns this is ready for review

**Comment by tristantarrant** [comment]
Please add rows to the table in `documentation/src/main/asciidoc/topics/ref_redis_commands.adoc`

**Comment by karesti** [comment]
@jabolina this is ready for review

**Comment by karesti** [comment]
@jabolina rebased from main

File: server/resp/src/main/java/org/infinispan/server/resp/ByteBufferUtils.java [hunk 1]
```diff
@@ -56,6 +59,48 @@ public static ByteBuf bytesToResult(byte[] result, ByteBufPool alloc) {
       return buffer;
    }
 
+   public static ByteBuf bytesToResult(List<byte[]> results, ByteBufPool alloc) {
```
Comment (jabolina) [inline server/resp/src/main/java/org/infinispan/server/resp/ByteBufferUtils.java:62]: Reeeeally nitpicking here. Regarding the cast of the `Collection` to a `List` on `POP`, so feel free to ignore this one. 

If this and the other `bytesToResult` receive a `Collection<byte[]>` argument, the cast on the `POP` can be avoided, and a `Immutables.immutableListConvert` or similar could be applied instead.

**Review by jabolina — COMMENTED** [review]
Looking good :green_circle:

I added a few suggestions, and feel free to skip the ones I am nitpicking.

File: multimap/src/main/java/org/infinispan/multimap/impl/EmbeddedMultimapListCache.java [hunk 2]
```diff
@@ -129,8 +131,40 @@ public CompletionStage<Collection<V>> subList(int from, int to) {
...
+    * or null if the key does not exit
+    */
+   public CompletionStage<Collection<V>> pollLast(K key, long count) {
+      return poll(key, count, false);
+   }
+
+   private CompletableFuture<Collection<V>> poll(K key, long count, boolean first) {
+      requireNonNull(key, "key can't be null");
+      requirePositive(count, "count can't be negative");
+      return readWriteMap.eval(key, new PollFunction<>(first, count));
```
[the remark this answers is not in the document]
Comment (karesti) [reply multimap/src/main/java/org/infinispan/multimap/impl/EmbeddedMultimapListCache.java:161]: There is a different  pop  for 0 for existing and asking for 0 for non existing.
if the key exists and we pop 0, the result is empty list
if the does not key exists and we pop 0, the result is null.

**Comment by karesti** [comment]
@jabolina changes reported

**Review by jabolina — APPROVED** [review]

**Comment by jabolina** [comment]
Integrated into main, thanks @karesti!

**Pull request merged**